cli/azd/pkg/project/framework_service_dotnet.go

```diff
@@ -42,16 +42,37 @@ func (dp *dotnetProject) Package(ctx context.Context, progress chan<- string) (s
 }
 
 func (dp *dotnetProject) InstallDependencies(ctx context.Context) error {
-	dotnetCli := tools.NewDotNetCli()
-	if err := dotnetCli.Restore(ctx, dp.config.Path()); err != nil {
+	if err := dp.dotnetCli.Restore(ctx, dp.config.Path()); err != nil {
 		return err
 	}
 	return nil
 }
 
+func (dp *dotnetProject) Initialize(ctx context.Context) error {
+	if err := dp.dotnetCli.InitializeSecret(ctx, dp.config.Path()); err != nil {
+		return err
+	}
+
+	handler := func(ctx context.Context, args ServiceLifecycleEventArgs) error {
+		bicepOutput := args.Args["bicepOutput"].(map[string]tools.AzCliDeploymentOutput)
```

Comment:
Recommend adding some error checking after attempting to cast your arg to the expected type.